google/cloud/forseti/scanner/scanners/iap_scanner.py

```diff
@@ -111,12 +111,27 @@ def instance_group_network_port(backend_service, instance_group):
         Returns:
             NetworkPort: how the service communicates with backends
         """
-        port = int(backend_service.port)
-        if backend_service.port_name:
-            for named_port in instance_group.named_ports or []:
-                if named_port.get('name') == backend_service.port_name:
-                    port = int(named_port.get('port'))
-                    break
+
+        # Field 'port' from backend service has been deprecated in favor of
+        # portName. PortName is required when the load balancing scheme is
+        # EXTERNAL. When the load balancing scheme is INTERNAL, this field
+        # is not used, it has the same behavior of port so we can just use
+        # portName to get the port from instance group.
+
+        if not backend_service.port_name:
+            # If backend_service.port_name is None, it's INTERNAL.
+            return None
+
+        port = None
```

Comment:
At this point, since we do have a valid backend service, do we not want to instantiate port as:
```
port = int(backend_service.port)
```

Reply:
The field port is deprecated, we shouldn't be using it in our code.

You can find more details here:
https://cloud.google.com/compute/docs/reference/rest/v1/backendServices

Reply:
Could there be any legacy issue?  i.e. some customer who are configured this from before and have not migrated to the new config / field?